python/__init__.py

```diff
@@ -473,84 +513,102 @@ def has_canfd(self) -> bool:
 
-  def get_usb_serial(self):
+  def get_usb_serial(self) -> Optional[str]:
     return self._serial
 
-  def get_secret(self):
+  def get_secret(self) -> bytearray:
+    assert self._handle is not None
     return self._handle.controlRead(Panda.REQUEST_IN, 0xd0, 1, 0, 0x10)
 
   # ******************* configuration *******************
 
-  def set_usb_power(self, on):
+  def set_usb_power(self, on: bool) -> None:
+    assert self._handle is not None
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xe6, int(on), 0, b'')
 
-  def set_power_save(self, power_save_enabled=0):
+  def set_power_save(self, power_save_enabled: int = 0) -> None:
+    assert self._handle is not None
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xe7, int(power_save_enabled), 0, b'')
 
-  def enable_deepsleep(self):
+  def enable_deepsleep(self) -> None:
+    assert self._handle is not None
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xfb, 0, 0, b'')
 
-  def set_esp_power(self, on):
+  def set_esp_power(self, on: bool) -> None:
+    assert self._handle is not None
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xd9, int(on), 0, b'')
 
-  def esp_reset(self, bootmode=0):
+  def esp_reset(self, bootmode: int = 0) -> None:
+    assert self._handle is not None
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xda, int(bootmode), 0, b'')
     time.sleep(0.2)
 
-  def set_safety_mode(self, mode=SAFETY_SILENT, param=0, disable_checks=True):
+  def set_safety_mode(self, mode: int = SAFETY_SILENT, param: int = 0, disable_checks: bool = True) -> None:
+    assert self._handle is not None
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xdc, mode, param, b'')
     if disable_checks:
       self.set_heartbeat_disabled()
       self.set_power_save(0)
 
-  def set_gmlan(self, bus=2):
+  def set_gmlan(self, bus: int = 2) -> None:
+    assert self._handle is not None
     # TODO: check panda type
     if bus is None:
       self._handle.controlWrite(Panda.REQUEST_OUT, 0xdb, 0, 0, b'')
     elif bus in (Panda.GMLAN_CAN2, Panda.GMLAN_CAN3):
       self._handle.controlWrite(Panda.REQUEST_OUT, 0xdb, 1, bus, b'')
 
-  def set_obd(self, obd):
+  def set_obd(self, obd: bool) -> None:
+    assert self._handle is not None
     # TODO: check panda type
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xdb, int(obd), 0, b'')
 
-  def set_can_loopback(self, enable):
+  def set_can_loopback(self, enable: bool) -> None:
+    assert self._handle is not None
     # set can loopback mode for all buses
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xe5, int(enable), 0, b'')
 
-  def set_can_enable(self, bus_num, enable):
+  def set_can_enable(self, bus_num: int, enable: bool) -> None:
+    assert self._handle is not None
     # sets the can transceiver enable pin
     self._handle.controlWrite(Panda.REQUEST_OUT, 0xf4, int(bus_num), int(enable), b'')
 
-  def set_can_speed_kbps(self, bus, speed):
+  def set_can_speed_kbps(self, bus: int, speed: Union[int, float]) -> None:
```

Comment:
there are a few cases where i type `Union[int, float]` since the code will technically convert it to int - search for that type & let me know if you'd like these changed to `int` only